inference-engine/src/mkldnn_plugin/nodes/roll.cpp

```diff
@@ -0,0 +1,318 @@
+            }
+
+            const auto axesTensorRank = axesTensor.getDims().size();
+            if (axesTensorRank > 1) {
+                IE_THROW() << "Roll layer with name '" << layerName << "' doesn't support 'axes' input tensor with rank: " << axesTensorRank;
+            }
+
+            /* Shift */
+            const auto& shiftTensor = layer->insData[SHIFT_INDEX].lock()->getTensorDesc();
+            const auto& shiftTensorPrec = layer->insData[SHIFT_INDEX].lock()->getTensorDesc().getPrecision();
+            if (shiftTensorPrec != Precision::I32 && shiftTensorPrec != Precision::I64) {
+                IE_THROW() << "Roll layer with name '" << layerName << "' has unsupported 'shift' input precision: " << shiftTensorPrec.name();
+            }
+
+            const auto shiftTensorRank = shiftTensor.getDims().size();
+            if (shiftTensorRank > 1) {
+                IE_THROW() << "Roll layer with name '" << layerName << "' doesn't support 'shift' input tensor with rank: " << shiftTensorRank;
+            }
+
+            m_shape = dataShape;
+
+            LayerConfig config;
+            for (size_t i = 0; i < layer->insData.size(); i++) {
+                DataConfig inConfig;
+                inConfig.inPlace = -1;
+                inConfig.constant = false;
+
+                Precision inPrecision = i > DATA_INDEX ? Precision(Precision::I32) : layer->insData[i].lock()->getTensorDesc().getPrecision();
+                if (inPrecision == Precision::BF16)
+                    inPrecision = Precision::FP32;
+                const SizeVector& inDims = layer->insData[i].lock()->getTensorDesc().getDims();
+                inConfig.desc = TensorDesc(inPrecision, inDims, InferenceEngine::TensorDesc::getLayoutByDims(inDims));
+                config.inConfs.push_back(inConfig);
+            }
+            DataConfig outConfig;
+            outConfig.inPlace = -1;
+            outConfig.constant = false;
+            Precision outPrecision = layer->insData[DATA_INDEX].lock()->getTensorDesc().getPrecision();
+            if (outPrecision == Precision::BF16) {
+                outPrecision = Precision::FP32;
+            }
+            const SizeVector& outDims = layer->outData[0]->getTensorDesc().getDims();
+            outConfig.desc = TensorDesc(outPrecision, outDims, InferenceEngine::TensorDesc::getLayoutByDims(outDims));
+
+            config.outConfs.push_back(outConfig);
+
+            config.dynBatchSupport = false;
+            confs.push_back(config);
+        } catch (InferenceEngine::Exception &ex) {
+            errorMsg = ex.what();
+        }
+    }
+
+    StatusCode execute(std::vector<Blob::Ptr> &inputs, std::vector<Blob::Ptr> &outputs, ResponseDesc *resp) noexcept {
+        m_shape = inputs[DATA_INDEX]->getTensorDesc().getDims();
+        m_numOfDims = m_shape.size();
+        m_shifts = extract1dIntegerDataFromBlob(inputs[SHIFT_INDEX]);
+        m_axes = extract1dIntegerDataFromBlob(inputs[AXES_INDEX]);
```

Comment:
No need to spend cpu time on excessive conversions. You've fixed precision - I32. Just use blobs from inputs.